community/CONTRIBUTING.md

```diff
@@ -5,6 +5,195 @@ Thanks for considering contributing to OPEA project. The contribution process is
+    - Dockerfile: `Dockerfile_[vendor]_[hardware]`
+    - Docker Compose: `docker_compose_[microservice type]_[vendor]_[hardware]` all lower case (i,e opea/llm-vllm-intel-hpu, opea/llm-faqgen-tgi-intel-hpu-svc)
+    - Docker Image: `opea/[microservice type]-[microservice sub type]-[library name]-[vendor]-[hardware]:latest`
+
+  - `tests/` : test should validate building image, starting service, validating microservie, stoping and destroying the container. Please refer to an example [test_asr_whisper.sh](https://github.com/opea-project/GenAIComps/blob/main/tests/asr/test_asr_whisper.sh). Please follow naming convention:`test_[microservice type]-[microservice sub type]-[library name]-[vendor]-[hardware].sh` 
+  - `README.md`: at minimum it should include: microservice description, build and start microservice commands and curl command with expected output.
+
+4. Now you have created all the required files, and validated your service. Last step is to modify the `README.md` at the component level `GenaIComps/comp/microservice type` to list your new component. Now you are ready to file your PR!. In the upcoming release maintainers will push the Docker Image to the Docker Hub. 
+
+5. After your component has been merged you are likely interested to build an application with it, and perhaps contributing it also to OPEA! so please continue to the "Contribute a GenAIExample" guide
+
+### Contribute a GenAIExample
+
+OPEA GenAIExamples examples offer simple deployment, testing, and scalability with Docker compose and Kubernetes. when contributing an example Docker compose deployment is the minimum requirement, but given OPEA is intended for enterprise applications, support for kubernetes deployment is highly encouraged.
+
+- Navigate to [OPEA GenAIExamples](https://github.com/opea-project/GenAIExamples/tree/main) and check the catalog of examples. If you find one that is very similar to what you are looking for, you can contribute your variation of it to that particular example folder. If you are bringing a completly new application you will need to create a separate example folder. 
+
+- Before stitching together all the microservices to build your application, let's make sure all the required building blocks are available!. You will notice [ChatQnA Flow Chart]() OPEA uses gateways to handle requests and route them to the corresponding megaservices (unless you have an agent that will otherwise handle the gateway function). If you are just making small changes to the application, like swaping one DB for another, you can reuse the existng Gateway but if you are contributing a completely new application, you will need to add a Gateway class. Navigate to [OPEA GenAIComps Gateway](https://github.com/opea-project/GenAIComps/blob/main/comps/cores/mega/gateway.py) and implement how Gateway should handle requests for your application. Note that Gateway implementation will move to GenAIExamples in future release. 
```

Comment:
[ChatQnA Flow Chart]()? put a link?

Reply:
I don't know how to put a link to a mermaid flowchart so i added link to the README that constains it @dbkinder do you know how to do that?

Reply:
Yes I do :)

Reply:
@dbkinder please add it!